### PR TITLE
chore: deep link documentation for registering proposals

### LINKED ISSUE #4872: [Task]: Write deep link documentation for registering events
### Task description

In order to see non-official Governance proposals on your dashboard, you have to "register" them via deep links. The specification for these deep links (as well as Firefly's other deep links) has **NOT** been written yet.

This is important for platforms like Soonaverse that would like an easier integration to interact with Firefly. It may be worth thinking for the future about an externally facing library both us and the community can use for parsing and building deep links.

### Requirements

- [x] Add external doc (somewhere? wiki?) for Firefly deep links

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

We desperately needed documentation for our deep link design, so that it can be more easily used by any platform wishing to interact with Firefly.

## Changelog

```
- Add deep link specification
- Add images of some UI (for new doc)
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Closes #4872

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

1. Checkout this branch (`chore/deep-link-documentation-for-registering-events`)
2. Run `yarn docs:start` in the root directory of the repository 

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: docs/specifications/deep-links.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+---
+icon: link
+---
+
+# Deep Links
+
+Deep links are special URLs that, when navigated to, open
+applications rather than a website in a browser.
```
Comment (Tuditi) [inline docs/specifications/deep-links.md:8]:
```suggestion
applications rather than a website.
```

File: docs/specifications/deep-links.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+The parameters are as follows:
+
+- `stage` - indicates a specific stage of the app to target, options are:
+  - `alpha` - the first available version of Firefly containing brand new features
+  - `beta` - the next available version of Firefly containing new but slightly tested features
+  - `shimmer` - the Firefly Shimmer version, containing new and well-tested features 
+- `context` - the part of Firefly that contains the operation, options are:
+  - `wallet` - managing coins and tokens
+  - `collectibles` - managing NFTs
```
Comment (Tuditi) [inline docs/specifications/deep-links.md:39]: does 'nft' make more sense maybe?

File: docs/specifications/deep-links.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+The following parameters are **required**:
+
+- `address` - the recipient's address where the funds will be sent to
+  - **MUST** be a Bech32 address; considering support for other address types in the future
+- `amount` - the amount of tokens to send in the transaction
+  - **MAY** contain a decimal so long as it makes sense given the value of the `unit` param (see below)
+
+The following parameters are **optional**:
+
+- `unit` - a specified denomination of the token to use, if applicable (default for IOTA is `Mi`, SMR is `SMR`)
```
Comment (Tuditi) [inline docs/specifications/deep-links.md:81]: And then we can delete this. I have a strong opinion on this, but I might be wrong. So if you feel strongly opposed, feel free to convince me :D

File: docs/specifications/deep-links.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+- `surplus` - send additional amounts of the base token when transferring native tokens
+
+Example:
+
+[!button Click me!](firefly://wallet/sendForm?address=iota1qrhacyfwlcnzkvzteumekfkrrwks98mpdm37cj4xx3drvmjvnep6xqgyzyx&amount=10&unit=Gi&giftStorageDeposit=true&surplus=1&metadata=Take%20my%20money)
+
+Source:
+
+```
+firefly://wallet/sendConfirmation?address=iota1qrhacyfwlcnzkvzteumekfkrrwks98mpdm37cj4xx3drvmjvnep6xqgyzyx&amount=10&unit=Gi&giftStorageDeposit=true&disableToggleGift=true&surplus=1&metadata=Take%20my%20money
```
Comment (Tuditi) [inline docs/specifications/deep-links.md:135]: you should put your own smr address here :rofl:

**Review by Tuditi — COMMENTED** [review]
Great work! :partying_face:

File: docs/specifications/deep-links.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+we are interested in the user experiences that they enable
+between websites, applications, platforms, etc. by providing more interoperability.
+
+Firefly has its own deep link scheme, exposing (limited) functionality that is required in 
+some type of user flow. A trivial example would be a user is buying native tokens on Soonaverse and must make a payment transaction
+in order to execute the buy order. Clicking on a deep link embedded inside the Soonaverse platform will trigger Firefly to 
+open and auto-fill the transaction data as necessary, making it simple confirm and click job for the user. 
+
+:::caution
+Firefly **will NEVER** automatically execute actions initiated by a deep link; they should **ALWAYS** require manual
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply docs/specifications/deep-links.md:18]: I think this softens the language in a way that we don't want. Maybe add "they will **ALWAYS**..."?

File: docs/specifications/deep-links.md (hunk 2 above)
Comment (maxwellmattryan) [reply docs/specifications/deep-links.md:38]: This is supposed to be a 1:1 mapping with the contexts of the app, so I think `collectibles` would be better.

File: docs/specifications/deep-links.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+```
+firefly://wallet/sendForm?address=<address>&amount=<amount>[&unit=<unit>][&assetId=<assetId>][&metadata=<metadata>][&tag=<tag>]
+```
+
+The following parameters are **required**:
+
+- `address` - the recipient's address where the funds will be sent to
+  - **MUST** be a Bech32 address; considering support for other address types in the future
+- `amount` - the amount of tokens to send in the transaction
+  - **MAY** contain a decimal so long as it makes sense given the value of the `unit` param (see below)
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply docs/specifications/deep-links.md:76]: That sounds good to me! The only problem is that we'll need to update the implementation / logic to reflect that then also notify relevant people of a breaking change in the scheme.

I think perhaps we can merge this version in the handbook but leave the other in the wiki so devs going to the wiki will see the source of truth wrt to what's actually implemented.

File: docs/specifications/deep-links.md (hunk 4 above)
Comment (maxwellmattryan) [reply docs/specifications/deep-links.md:134]: Oh I'll do it :joy:

File: docs/specifications/deep-links.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Tuditi) [reply docs/specifications/deep-links.md:77]: Sounds good, I'll create an issue to track this!

**Review by Tuditi — APPROVED** [review]
LGTM! :fire:

**Pull request merged**